### PR TITLE
Reindex resume from status

### PR DESCRIPTION
**Pull request by henningandersen**
Reindex now uses last known good status on failure to ensure that counts
continue from where it left when a node fails.

Also added `persistentTaskId` to `StartReindexResponse` and removed a
couple of obsolete todos.

Relates #42612

### CODE REVIEW

File: server/src/main/java/org/elasticsearch/index/reindex/WorkerBulkByScrollTaskState.java [hunk 1]
```diff
@@ -244,6 +248,25 @@ public void rethrottle(float newRequestsPerSecond) {
         }
     }
 
+    public void resetStatus(BulkByScrollTask.Status checkpointStatus) {
```
Comment (tbrooks8) [inline server/src/main/java/org/elasticsearch/index/reindex/WorkerBulkByScrollTaskState.java]: Maybe `initStatus`? or `setInitialStatus`?

File: modules/reindex/src/main/java/org/elasticsearch/index/reindex/StartReindexJobAction.java [hunk 2]
```diff
@@ -95,45 +95,55 @@ public boolean getWaitForCompletion() {
...
             out.writeString(taskId);
+            out.writeString(persistentTaskId);
             out.writeOptionalWriteable(reindexResponse);
         }
 
         public String getTaskId() {
             return taskId;
         }
 
+        public String getPersistentTaskId() {
```
Comment (tbrooks8) [inline modules/reindex/src/main/java/org/elasticsearch/index/reindex/StartReindexJobAction.java]: I was just working on this on a different branch and I feel like the persistent task id should be the default naming? As in the methods are `getTaskId` or  `getReindexTaskId`? And the other one be `getEphemeralTaskId`? Or `getCoordinatorNodeTaskId`? Or `getInitialEphemeralTaskId`?

**Review by Tim-Brooks — COMMENTED** [review]
Mostly looks good, just some questions about naming that we can discuss tomorrow if necessary.

**Comment by henningandersen** [comment]
@elasticmachine run elasticsearch-ci/packaging-sample-matrix-unix

**Review by Tim-Brooks — APPROVED** [review]
LGTM

**Pull request merged**